pkg/providerConfigs/detokenize.go

```diff
@@ -7,308 +7,197 @@ See the LICENSE file for more details.
 package providerConfigs
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"github.com/Masterminds/sprig/v3"
+	"io"
 	"os"
 	"path/filepath"
-	"strconv"
+	"reflect"
+	"regexp"
 	"strings"
-
-	log "github.com/rs/zerolog/log"
+	"text/template"
 )
 
-// DetokenizeGitGitops - Translate tokens by values on a given path
-func DetokenizeGitGitops(path string, tokens *GitopsDirectoryValues, gitProtocol string, useCloudflareOriginIssuer bool) error {
-	err := filepath.Walk(path, detokenizeGitops(path, tokens, gitProtocol, useCloudflareOriginIssuer))
+type TerraformDecoderFunc func(content string, dest []byte) error
+
+// ToTemplateVars - converts a string to a template variable
+func ToTemplateVars(input string, instance Tokens) string {
+	value := reflect.ValueOf(instance)
+	if value.Kind() == reflect.Ptr {
+		value = value.Elem()
+	}
+	sanitizer := strings.NewReplacer("<", "", "_", "", ">", "", "-", "")
+
+	fields := value.Type()
+	normalizedName := strings.ToLower(sanitizer.Replace(input))
+	for i := 0; i < fields.NumField(); i++ {
+		field := fields.Field(i)
+		val := value.Field(i)
+
+		if normalizedName == strings.ToLower(field.Name) {
+			// If the field value is the zero value of the field template
+			// will add - to the template variable to clean up extra whitespace.
+			// We do not have any bool values in the tokens, so we should be ok.
+			// TODO: Find a better way to handle this.
+			if val.IsZero() {
+				return fmt.Sprintf("%s- .%s %s", leftDelimiter, field.Name, rightDelimiter)
+			}
+			// if field name matches return the correct formatted name
+			return fmt.Sprintf("%s .%s %s", leftDelimiter, field.Name, rightDelimiter)
+		}
+	}
+
+	// If no match found, return the original input as a string, so we can have a visual indication
+	// that the token was not found in the Token Struct without erroring out
+	return "<variable-not-found>"
+}
+
+// Detokenize - Translate tokens by values on a given path
+func Detokenize(path string, tokens Tokens, gitProtocol string, useCloudflareOriginIssuer bool) error {
+	err := filepath.Walk(path, detokenize(path, tokens, gitProtocol, useCloudflareOriginIssuer))
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func detokenizeGitops(path string, tokens *GitopsDirectoryValues, gitProtocol string, useCloudflareOriginIssuer bool) filepath.WalkFunc {
+func detokenize(path string, tokens Tokens, gitProtocol string, useCloudflareOriginIssuer bool) filepath.WalkFunc {
 	return filepath.WalkFunc(func(path string, fi os.FileInfo, err error) error {
```

Comment:
Seems to have been there from before but the `path string` from `func detokenize()` is never used, since the `path string` inside `filepath.WalkFunc` is not the same as the top one.